src/main/java/dk/kb/util/YAMLUtils.java

```diff
@@ -0,0 +1,61 @@
+package dk.kb.util;
+
+import java.util.Map;
+import java.util.Objects;
+import java.util.Properties;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+
+public class YAMLUtils {
+    
+    /**
+     * Converts the YAML object into a properties object
+     *
+     * @param collection the yaml
+     * @return the yaml as properties
+     */
+    public static Properties toProperties(YAML collection) {
+        Map<String, Object> propertiesMap = toFlatList(collection, null)
+                                                    .collect(Collectors.toMap(
+                                                            tuple2 -> tuple2._1,
+                                                            tuple2 -> tuple2._2
+                                                    ));
+        
+        Properties properties = new Properties();
+        
+        properties.putAll(propertiesMap);
+        return properties;
+    }
+    
+    protected static Stream<Tuple2<String, Object>> toFlatList(Map<String, Object> collection, String keyFix) {
+    
+        return collection.entrySet().stream().flatMap(
+                entry -> {
+                    String key = Stream.of(keyFix,entry.getKey()).filter(Objects::nonNull).collect(Collectors.joining("."));
```

Comment:
"_When you have a hammer, everything looks like a nail_". What about
```
String key = keyFix == null ? entry.getKey() : keyFix + "." + entry.getKey();
```
The stream-hack is a hack since keyFix is a `String`. If it was a `List<String>` used to preserve the individual steps it might make sense.

Reply:
Accepted